Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp

```diff
@@ -0,0 +1,255 @@
+            {
+                filteredSourceInfos.Append(sourceInfoIter.Current());
+            }
+            sourceInfoIter.MoveNext();
+        }
+        sourceGroupIter.MoveNext();
+    }
+    return filteredSourceInfos;
+}
+
+int GetSGSelection(IVector<MediaFrameSourceInfo> filteredGroup)
+{
+    unsigned int idx = 0;
+    auto group = filteredGroup.First();
+    while (group.HasCurrent())
+    {
+        idx++;
+        auto currGroup = group.Current();
+        if (currGroup.SourceGroup() == nullptr)
+        {
+            std::cout << "nullptr";
+        }
+        // These are in the same order as the enum MediaStreamType
+        // TODO: better solution is to create a Pair type lookup which is future proof
+        std::wstring streamTypes[] = 
+        { 
+            L"VideoPreview", 
+            L"VideoRecord", 
+            L"Audio", 
+            L"Photo" 
+        };
+
+        // These are in the same order as the enum Panel
+        std::wstring panelTypes[] = 
+        {
+            L"Unknown",
+            L"Front",
+            L"Back",
+            L"Top",
+            L"Bottom",
+            L"Left",
+            L"Right"
+        };
+        auto enclosureLocation = currGroup.DeviceInformation().EnclosureLocation();
+        std::wstring panelLocation;
+        if (enclosureLocation)
+        {
+            panelLocation = panelTypes[(int)enclosureLocation.Panel()];
+        }
+        
+        std::wcout << idx << L":" << currGroup.DeviceInformation().Name().c_str() << ":" << streamTypes[(int)currGroup.MediaStreamType()].c_str() << panelLocation.c_str() << std::endl;
+        group.MoveNext();
+    }
+    do
+    {
+        std::wcout << L"Enter selection:";
+        std::cin >> idx;
+    } while ((idx > filteredGroup.Size()) || (idx < 1));
+    return idx-1;
+}
+
+int GetMediaTypeSelection(MediaFrameSourceInfo selectedSource)
+{
+    auto mediaDescriptionIter = selectedSource.VideoProfileMediaDescription().First();
+    int idx = 0;
+    while (mediaDescriptionIter.HasCurrent())
+    {
+
+        auto format = mediaDescriptionIter.Current();
+        int frameRateInt = (int)format.FrameRate();
+        int frameRateDec = ((int)(format.FrameRate() * 100) % 100);
+        idx++;
+        auto formatStr = std::to_wstring(idx) + L". " + std::wstring(format.Subtype().c_str()) + L": " +
+            std::to_wstring(format.Width()) + L"x" + std::to_wstring(format.Height()) + L" @ " + std::to_wstring(frameRateInt) + L"." + std::to_wstring(frameRateDec) + L" fps";
+        std::wcout << formatStr << std::endl;
+        mediaDescriptionIter.MoveNext();
+    }
+    int selection;
+    do
+    {
+        std::wcout << L"Enter selection:";
+        std::cin >> selection;
+    } while ((selection > idx) || (selection < 1));
+    return selection-1; // because selection idx is from 1 to N and indexes for from 0 to N-1
+}
+
+void TakePhotosAndProcess(MediaCapture mediaCapture, uint32_t photoIndex)
+{

```

Comment:
add a try and catch, these APIs can fail. You may want to have a master try-catch as well in your main that outputs meaningful error message